app/public/components/filter/filteringPanel.js

```diff
@@ -0,0 +1,128 @@
+        }
+
+        if (leftFilterInput) {
+            model.filtering.addFilter(fieldNameValue, leftFilterInput, leftFilterType);
+        }
+
+        clearUserInput();
+    };
+
+    return h('.font-size-small', [
+        h('.flex-wrap.justify-between.items-center',
+            h('.flex-wrap.justify-between.items-center',
+                h('h5', 'Filter data'),
+                h('button.btn.btn-secondary', {
+                    onclick: () => {},
+                }, 'Use defined filters'))),
+        h('.flex-wrap.justify-between.items-center',
+            h('.flex-wrap.justify-between.items-center',
+                h('select.select', {
+                    id: 'show-options-field',
+                    name: 'showOptions' }, [
+                    h('option', {
+                        value: '',
+                        selected: true,
+                        disabled: true,
+                        hidden: true,
+                    }, 'Field'),
+                    fields.map((field) => h('option', { value: field.name }, field.fieldName)),
+                ], h('.close-10-primary')),
+
+                h('select.select', {
+                    id: 'filters-opts-select',
+                    name: 'showOptions',
+                    onchange: () => onFilteringTypeChange(),
+                }, [
+                    h('option', { value: '', selected: true, disabled: true, hidden: true }, 'Filtering type'),
+                    h('option', { value: aggregatedFiltersTypes }, aggregatedFiltersTypes),
+                    h('option', { value: filterTypes.between }, filterTypes.between),
+                ], h('.close-10-primary')),
+
+                h('.text-field',
+                    h('input.form-control.relative', {
+                        style: 'width:120px',
+                        type: 'text',
+                        value: document.getElementById('filters-opts-select')?.options[Selection.selectedIndex]?.value,
+                        disabled: false,
+                        id: 'left-filter-input',
+                        required: true,
+                    }),
+                    h('span.placeholder', { id: 'left-filter-placeholder' }, 'match/from')),
+
+                h('.text-field',
+                    h('input.form-control.relative', {
+                        style: 'width:120px',
+                        type: 'text',
+                        value: '',
+                        disabled: false,
+                        id: 'right-filter-input',
+                        required: true,
+                    }),
+                    h('span.placeholder', { id: 'right-filter-placeholder' }, 'to/exclude')),
+
+                h('button.btn.btn-primary', {
+                    onclick: () => onFilterSubmit(),
+                }, 'Filter'),
+
+                h('button.btn.btn-secondary.icon-only-button', {
+                    onclick: () => onFilterSubmit(),
+                }, h('.save-20-primary')))),
+    ]);
```

Comment:
Splitting it to a few parts will make it more readable

Reply:
Done